docs/recovery.md

```diff
@@ -0,0 +1,351 @@
+```
+Input wallet password:
+Confirm wallet password:
+```
+
+nce a new wallet password has been obtained, the user will be prompted for
+their _existing_ cipher seed:
+```
+Input your 24-word mnemonic separated by spaces: ability noise lift document certain month shoot perfect matrix mango excess turkey river pitch fluid rack drill text buddy pool soul fatal ship jelly
+```
+
+If a _cipher seed passphrase_ was used when the seed was created, it MUST be entered now:
+```
+Input your cipher seed passphrase (press enter if your seed doesn't have a passphrase):
+```
+
+Finally, the user has an option to choose a _recovery window_:
+```
+Input an optional address look-ahead used to scan for used keys (default 2500):
+```
+
+The recovery window is a metric that the on-chain rescanner will use to
+determine when all the "used" addresses have been found. If the recovery window
+is 2, then after we find to consecutive addresses that haven't yet been found,
+then we'll exit. If an `lnd` on-chain wallet was extensively used, then users
+may want to _increase_ the default value.  
+
+If all the information provided was valid, then you'll be presented with the
+seed again: 
+```
+
+!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO RESTORE THE WALLET!!!
+
+---------------BEGIN LND CIPHER SEED---------------
+ 1. ability   2. noise   3. lift     4. document
+ 5. certain   6. month   7. shoot    8. perfect
+ 9. matrix   10. mango  11. excess  12. turkey
+13. river    14. pitch  15. fluid   16. rack
+17. drill    18. text   19. buddy   20. pool
+21. soul     22. fatal  23. ship    24. jelly
+---------------END LND CIPHER SEED-----------------
+
+!!!YOU MUST WRITE DOWN THIS SEED TO BE ABLE TO RESTORE THE WALLET!!!
+
+lnd successfully initialized!
+```
+
+In `lnd`'s logs, you should see something along the lines of (irrelevant lines skipped):
+```
+[INF] LNWL: Opened wallet
+[INF] LTND: Wallet recovery mode enabled with address lookahead of 2500 addresses
+[INF] LNWL: RECOVERY MODE ENABLED -- rescanning for used addresses with recovery_window=2500
+[INF] CHBU: Updating backup file at test_lnd3/data/chain/bitcoin/simnet/channel.backup
+[INF] CHBU: Swapping old multi backup file from test_lnd3/data/chain/bitcoin/simnet/temp-dont-use.backup to test_lnd3/data/chain/bitcoin/simnet/channel.backup
+[INF] LNWL: Seed birthday surpassed, starting recovery of wallet from height=748 hash=3032830c812a4a6ea305d8ead13b52e9e69d6400ff3c997970b6f76fbc770920 with recovery-window=2500
+[INF] LNWL: Scanning 1 blocks for recoverable addresses
+[INF] LNWL: Recovered addresses from blocks 748-748
+[INF] LNWL: Started rescan from block 3032830c812a4a6ea305d8ead13b52e9e69d6400ff3c997970b6f76fbc770920 (height 748) for 0 addresses
+[INF] LNWL: Catching up block hashes to height 748, this might take a while
+[INF] LNWL: Done catching up block hashes
+[INF] LNWL: Finished rescan for 0 addresses (synced to block 3032830c812a4a6ea305d8ead13b52e9e69d6400ff3c997970b6f76fbc770920, height 748)
```

Comment:
will the number of addresses show 0?

Reply:
Nope, it's just that the test node used for this example didn't have any addresses.

Reply:
filledin 800